src/Common.am

```diff
@@ -45,6 +45,9 @@ DEFS += -DDEFAULT_CONFIG_FILE=\"$(DEFAULT_CONFIG_FILE)\" \
 CLEANFILES =
 check_PROGRAMS =
 TESTS =
+# A space-separated list of header files to be skipped by
+# testHeaders during "make check"
+EXCLUDE_FROM_HDR_TESTING =
 
```

Comment:
We have good coverage. What we are missing is:
- lib/rfcnb/*.h
- lib/smblib/*.h
- lib/sspi/*.h
- src/acl/external/kerberos_ldap_group/support.h
- src/adaptation/ecap/*.h
- header files used by helpers (e.g. src/auth/digest/eDirectory/digest_common.h, or src/auth/basic/NCSA/crypt_md5.h)
- src/tests/testAuth.h
- tools/squidclient/*.h

The coverage list was obtained by filtering lines  'ok - .*' in the output log for a complete test-suite build, and comparing with the list of headers as obtained by 'git grep'

Reply:
- ecap is missed because I don't have libecap on my system
- squidclient is actually tested - missed it in my query
- the lib/*/*.h , I don't think we want to test. These are leaf files, they do not map to our standard coding guidelines, for instance

In other words, I'd say we're good.
For full reference, here's the list of files that are not tested:

```
+lib/rfcnb/byteorder.h
+lib/rfcnb/rfcnb-common.h
+lib/rfcnb/rfcnb-error.h
+lib/rfcnb/rfcnb.h
+lib/rfcnb/rfcnb-io.h
+lib/rfcnb/rfcnb-priv.h
+lib/rfcnb/rfcnb-util.h
+lib/rfcnb/std-includes.h
+lib/smblib/md4.h
+lib/smblib/smbdes.h
+lib/smblib/smbencrypt.h
+lib/smblib/smblib-common.h
+lib/smblib/smblib.h
+lib/smblib/smblib-priv.h
+lib/smblib/std-defines.h
+lib/sspi/sspwin32.h
+src/acl/external/kerberos_ldap_group/support.h
+src/adaptation/ecap/Config.h
+src/adaptation/ecap/Host.h
+src/adaptation/ecap/MessageRep.h
+src/adaptation/ecap/Registry.h
+src/adaptation/ecap/ServiceRep.h
+src/adaptation/ecap/XactionRep.h
+src/auth/basic/NIS/nis_support.h
+src/auth/basic/SMB_LM/msntauth.h
+src/auth/basic/SMB_LM/valid.h
+src/auth/basic/SSPI/valid.h
+src/auth/digest/eDirectory/digest_common.h
+src/auth/digest/eDirectory/edir_ldapext.h
+src/auth/digest/eDirectory/ldap_backend.h
+src/auth/digest/LDAP/digest_common.h
+src/auth/digest/LDAP/ldap_backend.h
+src/auth/negotiate/kerberos/negotiate_kerberos.h
+src/tests/testAuth.h
```

Reply:
> the lib//.h , I don't think we want to test

... but this PR _is_ testing them now, right?! For example, lib/rfcnb/Makefile.am does include Common.am and does not exclude any lib/rfcnb headers from testing. Same for lib/smblib/Makefile.am.

> These are leaf files, they do not map to our standard coding guidelines, for instance

I do not know what you mean by "leaf" but many files in src/ also did not meet our guidelines (as far as header autonomy is concerned) until this PR fixed them. Perhaps we can adjust these files as well instead of making them exceptional?

Please do not get me wrong -- if there is a valid reason to exclude these files, then I am not against excluding them, especially if it is difficult to fix them.

Reply:
Right now they are not being tested because - at least on my system - none of the test-suite
options require them. In other words, it's working as intended.

These directories are entered if the Automake conditionals ENABLE_SMBLIB and ENABLE_SSPI are enabled,
which aren't because (at least for SSPI) they require a windows build via src/auth/ntlm/SSPI/required.m4 preconditions

Reply:
In the context of "I don't think we _want_ to test" phrase (emphasis mine), I was talking about what we want this PR to test in at least one build across all possible builds. It looks like you were talking about what a few specific PR builds in your specific environment do not test. If you agree that lib/rfcnb/ and lib/smblib/ headers are (expected to be) tested by this PR (in at least one build across all possible builds), then we are in agreement.

If you know of a header file that is not tested at least once across all possible PR builds across all possible environments, then let's discuss that exceptional case. I have updated PR description and am resolving this thread under the assumption that we are not aware of such header files.